src/rgw/rgw_trim_bilog.cc

```diff
@@ -571,13 +735,40 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
       return set_cr_error(retcode);
     }
 
+    retcode = maybe_remove_generation();
```

Comment:
this trimming logic is very conservative about limiting concurrent omap deletions, and `BucketCleanIndexCollectCR` in particular could be very expensive. `BucketTrimInstanceCR` should only call one of `BucketTrimShardCollectCR` or `BucketCleanIndexCollectCR`, but not both in the same trim interval

so i think we should do this check first, and only go on to call `take_min_status()`/`BucketTrimShardCollectCR` if there were no older generations to delete

Reply:
Done.

Reply:
Done.